### PR TITLE
Add logging/exception handling to controller

### PR DESCRIPTION
**Pull request by kjduensing**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
Adds logging and fixes a "not found" edge case

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/92256

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
Travel Pay only

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

File: modules/travel_pay/app/controllers/travel_pay/v0/claims_controller.rb [hunk 1]
```diff
@@ -37,19 +37,23 @@ def show
...
         end
 
         begin
-          appt = appts_service.get_appointment_by_date_time({ 'appt_datetime' => params['appointmentDatetime'] })
+          Rails.logger.info(message: 'SMOC transaction START')
 
-          claim = claims_service.create_new_claim({ 'btsss_appt_id' => appt[:data]['id'] })
-
-          claim_id = claim['claimId']
+          appt_id = get_appt_or_raise(params['appointmentDatetime'])
```
Comment (liztownd) [inline modules/travel_pay/app/controllers/travel_pay/v0/claims_controller.rb:47]: Nice! I like the addition of the handling for both these calls!

**Review by liztownd — APPROVED** [review]
Looks good!

**Review by seesleestak — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]
